src_py/sprite.py

```diff
@@ -110,7 +111,7 @@ class Sprite:
     """
 
     def __init__(self, *groups):
-        self.__g = {}  # The groups the sprite is in
+        self.__g = WeakSet()  # The groups the sprite is in
```

Comment:
Maybe WeakKeyDictionary?

Reply:
Why? Isn't `__g` a set?

Reply:
Bruh, I didn't even think about the fact that `{}` is a `dict`. I just assumed it was a set.

But now I am asking myself why this isn't a set?

Reply:
Sets weren't included with python back then, they were an external extension. So people mostly implemented sets with dicts.

Reply:
I expected that but I also thought that the code would have been updated since then.